src/components/pagination/docs.mdx

```diff
@@ -0,0 +1,101 @@
+---
+componentName: Pagination
+---
+
+## Overview
+
+The Pagination component allows users to navigate through a large dataset split into pages. These pages could be either links (different URLs) or content rendered within the same page/URL.
+
+It is not meant to be used as a navigation control for regular content or as a stepper, i.e. a guide, tutorial, setup or creation flow.
+
+### `Pagination`
+
+This is a container component that provides grid layout for the subcomponents and pagination react context.
+
+**Props:**
+
+- `totalItems` (required): Total number of items
+- `pageSize` (required): Number of items per page
+- `page` (optional): Current page index. This is 1-based, and is only used for the initial render. If not provided, the component will use the first page by default.
+- `onPageChange` (optional): Callback function to be called when a page is selected. It receives the page index as an argument.
+- `onPageSizeChange` (optional): Callback function to be called when a page size is selected. It receives the page size as an argument.
+
+### `Pagination.Info`
+
+Displays info like, _"90-100 of 103 items"_
+
+**Props:**
+
+- `showTotalItems`: Whether to show the total number of items. Default: `false``
+
+### `Pagination.Nav`
+
+Displays navigation buttons.
+
+**Props:**
+
+- `type`: "compact" | "numbered" | "truncated"
+  - "compact": Displays only previous and next buttons
+  - "numbered": Displays numbered buttons for each page
+  - "truncated": Displays numbered buttons for each page, but truncates the list if there are too many pages
+
+### `Pagination.SizeSelector`
+
+Displays a dropdown to select the page size.
+
+- `sizes`: Array of page size numbers
+
+## Basic
+
+<LiveComponent>
+	{`<Pagination
+	totalItems={103}
+	pageSize={10}
+	page={3}
+	onPageChange={(idx, pagesize) => console.log("Select page: %s; Pagesize: %s", idx, pagesize)}
+>
+	<Pagination.Info showTotalItems={true}/>
+	<Pagination.Nav type="truncated" />
+	<Pagination.SizeSelector sizes={[10, 30, 50]}/>
+</Pagination>
+`}
+</LiveComponent>
+
+## With Elements
```

Comment:
This demo does not actually render the appropriate items when I switch "Items per page" to anything other than 10.  It still will render the items as if you have 10 selected for me.

Reply:
Fixed here: https://github.com/hashicorp/dev-portal/pull/1508/commits/c7a6e0da64687ed18ebaafc1cd33328e3b354faa (no additional testcase needed as it was consumer error)

Reply:
Fixed in: https://github.com/hashicorp/dev-portal/pull/1508/commits/5180305452e80c31f12c8f5c1fcf6d4de7c4bcd7

Reply:
On initial load, the `Info` component isn't showing the correct numbers. It starts with `21 – 30 of 83` when I'd expect `1 – 30 of 83` since the current page is `1` and the page size is `30`.

![image](https://user-images.githubusercontent.com/43934258/211674254-b79c9856-6ad4-4f21-94fc-ce97c469273e.png)

Reply:
Good catch! This was a logic bug — will fix and add a test

Reply:
Fixed in: https://github.com/hashicorp/dev-portal/pull/1508/commits/77fbb35d914fffcdfc756ca5eaf87b21d56cbe94